mlos_bench/mlos_bench/service/types/config_loader_type.py

```diff
@@ -0,0 +1,110 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Protocol interface for helper functions to lookup and load configs.
+"""
+
+from typing import List, Optional, Union, Protocol, runtime_checkable, TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from mlos_bench.tunables.tunable_groups import TunableGroups
+    from mlos_bench.service.base_service import Service
+    from mlos_bench.environment.base_environment import Environment
+
+
+@runtime_checkable
+class SupportsConfigLoading(Protocol):
+    """
+    Protocol interface for helper functions to lookup and load configs.
+    """
+
+    def resolve_path(self, file_path: str,
+                     extra_paths: Optional[List[str]] = None) -> str:
+        """
+        Prepend the suitable `_config_path` to `path` if the latter is not absolute.
+        If `_config_path` is `None` or `path` is absolute, return `path` as is.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to the input config file.
+        extra_paths : List[str]
+            Additional directories to prepend to the list of search paths.
+
+        Returns
+        -------
+        path : str
+            An actual path to the config or script.
+        """
+
+    def load_config(self, json_file_name: str) -> Union[dict, List[dict]]:
+        """
+        Load JSON config file. Search for a file relative to `_config_path`
+        if the input path is not absolute.
+        This method is exported to be used as a service.
+
+        Parameters
+        ----------
+        json_file_name : str
+            Path to the input config file.
+
+        Returns
+        -------
+        config : Union[dict, List[dict]]
+            Free-format dictionary that contains the configuration.
+        """
+
+    def build_environment(self, config: dict,
+                          global_config: Optional[dict] = None,
+                          tunables: Optional["TunableGroups"] = None,
+                          service: Optional["Service"] = None) -> "Environment":
```

Comment:
do we need quotation here?
```suggestion
                          service: Optional[Service] = None) -> Environment:
```
etc.

Reply:
Works.  I guess let's go with that for now.  Can always change it back later if need be.

Reply:
Nevermind, I'd only run the `make check` target.  `make test` fails when I do it this way since "Service" imports `config_loader_type` which would import `Service` and Environment imports both.
Reverted back to what I had for now.